test/access/AccessControl.behavior.js

```diff
@@ -210,8 +211,293 @@ function shouldBehaveLikeAccessControlEnumerable(errorPrefix, admin, authorized,
+
+      beforeEach(async function () {
+        incorrectIncreaseTo = correctIncreaseTo.subn(1);
+      });
+
+      it('should revert if block.timestamp is equal to delayed until', async function () {
+        await time.increaseTo(incorrectIncreaseTo);
+        await expectRevert(
+          this.accessControl.acceptDefaultAdminTransfer({ from: correctPendingDefaultAdmin }),
+          `${errorPrefix}: delayedUntil not met`,
+        );
+      });
+
+      it('should revert if block.timestamp is less to delayed until', async function () {
+        await time.increaseTo(incorrectIncreaseTo.subn(1));
+        await expectRevert(
+          this.accessControl.acceptDefaultAdminTransfer({ from: correctPendingDefaultAdmin }),
+          `${errorPrefix}: delayedUntil not met`,
+        );
+      });
+    });
+  });
+
+  describe('cancel transfer default admin', async function () {
+    beforeEach(async function () {
+      await this.accessControl.beginDefaultAdminTransfer(newDefaultAdmin, { from: defaultAdmin });
+    });
+
+    it('resets pending default admin and delayed until', async function () {
+      await this.accessControl.cancelDefaultAdminTransfer({ from: defaultAdmin });
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(web3.utils.toBN(0));
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(ZERO_ADDRESS);
+    });
+
+    it('reverts if called by non default admin accounts', async function () {
+      await expectRevert(
+        this.accessControl.cancelDefaultAdminTransfer({ from: other }),
+        `${errorPrefix}: account ${other.toLowerCase()} is missing role ${DEFAULT_ADMIN_ROLE}`,
+      );
+    });
+  });
+
+  describe('renouncing admin', async function () {
+    let correctIncreaseTo;
+    let from = defaultAdmin;
+
+    beforeEach(async function () {
+      correctIncreaseTo = (await time.latest()).add(delay).addn(1);
+      await this.accessControl.beginDefaultAdminTransfer(ZERO_ADDRESS, { from });
+    });
+
+    describe('caller is default admin and delayed until is met', async function () {
+      let receipt;
+
+      beforeEach(async function () {
+        await time.increaseTo(correctIncreaseTo);
+        from = defaultAdmin;
+        receipt = await this.accessControl.renounceRole(DEFAULT_ADMIN_ROLE, from, { from });
+      });
+
+      it('renounces role and does not grant it to the ZERO ADDRESS', async function () {
+        expect(await this.accessControl.hasRole(DEFAULT_ADMIN_ROLE, defaultAdmin)).to.be.false;
+        expect(await this.accessControl.hasRole(ZERO_ADDRESS, defaultAdmin)).to.be.false;
+      });
+
+      it('emits events', async function () {
+        expectEvent(receipt, 'RoleRevoked', {
+          role: DEFAULT_ADMIN_ROLE,
+          account: from,
+        });
+      });
+
+      it('marks the contract as not owned', async function () {
+        expect(await this.accessControl.owner()).to.equal(ZERO_ADDRESS);
+      });
```

Comment:
These also look to me like they should be just one test... Now I'm not sure if what I'm saying goes against what we've done across the test suite in general.

Reply:
I think I put them separately because the last one that allows recovery via `_grantRole` used the same `before`, but it's cleaner the way you propose.

Added in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/ac093f42f13bbb6b52386c5c16212e3207d51ace